wagtail/contrib/modeladmin/tests/test_modeladmin_edit_handlers.py

```diff
@@ -0,0 +1,136 @@
+        self.assertIn('phone_number', response.content.decode('UTF-8'))
+        self.assertNotIn('address', response.content.decode('UTF-8'))
+        self.assertEqual(
+            [ii for ii in response.context['form'].fields],
+            ['first_name', 'last_name', 'phone_number']
+        )
+
+    def test_model_panels(self):
+        # tests that panel definition from model is used to create
+        # a model instance if no edit_handler is set on the model
+        response = self.client.post('/admin/modeladmintest/friend/create/', {
+            'first_name': "John",
+            'last_name': "Doe",
+            'address': "123 Main St Anytown",
+            'phone_number': "+123456789"
+        })
+        # Should redirect back to index
+        self.assertRedirects(response, '/admin/modeladmintest/friend/')
+
+        # Check that the friend was created
+        self.assertEqual(Friend.objects.filter(
+            first_name="John",
+            phone_number="+123456789"
+        ).count(), 1)
+
+        # verify that form fields are returned which have been defined
+        # in Friend.panels
+        response = self.client.get('/admin/modeladmintest/friend/create/')
+        self.assertIn('first_name', response.content.decode('UTF-8'))
+        self.assertNotIn('last_name', response.content.decode('UTF-8'))
+        self.assertIn('phone_number', response.content.decode('UTF-8'))
+        self.assertNotIn('address', response.content.decode('UTF-8'))
+        self.assertEqual(
+            [ii for ii in response.context['form'].fields],
+            ['first_name', 'phone_number']
+        )
+
+    def test_model_admin_edit_handler(self):
+        # tests that edit_handler definition from modeladmin is used to create
+        # a model instance if no edit_handler is set on the model
+        response = self.client.post('/admin/modeladmintest/visitor/create/', {
+            'first_name': "John",
+            'last_name': "Doe",
+            'address': "123 Main St Anytown",
+            'phone_number': "+123456789"
+        })
+        # Should redirect back to index
+        self.assertRedirects(response, '/admin/modeladmintest/visitor/')
+
+        # Check that the person was created
+        self.assertEqual(Visitor.objects.filter(
+            last_name="Doe",
+            phone_number="+123456789",
+            address="123 Main St Anytown"
+        ).count(), 1)
+
+        # verify that form fields are returned which have been defined
+        # in Person.edit_handler
+        response = self.client.get('/admin/modeladmintest/visitor/create/')
+
+        self.assertIn('last_name', response.content.decode('UTF-8'))
+        self.assertIn('phone_number', response.content.decode('UTF-8'))
+        self.assertIn('address', response.content.decode('UTF-8'))
+        self.assertNotIn('first_name', response.content.decode('UTF-8'))
+        self.assertEqual(
+            [ii for ii in response.context['form'].fields],
```

Comment:
@mr-stateradio A very small thing, but for readability's sake, I think it could be good to use a more descriptive variable name than `ii` here. `field_name` might make a good substitute.

Reply:
@ababic I agree and will fix it. I copied the structure and some variable names from `test_simple_modeladmin.TestCreateView.test_create`

Reply:
@ababic  fixed in https://github.com/wagtail/wagtail/pull/4900/commits/57f838b2393b145bfe1655b8e01b3e01e4bdd9c4